moderation/user_notes/cog.py

```
@@ -0,0 +1,59 @@
+from PyDrocsid.emojis import name_to_emoji
+from discord import Member, Embed
+
+from PyDrocsid.database import db, filter_by
+
+from PyDrocsid.config import Contributor
+from discord.ext import commands
+from discord.ext.commands import Context, UserInputError
+
+from PyDrocsid.cog import Cog
+from cogs.library.moderation.mod.permissions import ModPermission
+from cogs.library.moderation.user_notes.models import UserNote
+from PyDrocsid.translations import t
+from cogs.library.moderation.user_notes.permissions import UserNotePermissions
+
+t = t.user_notes
+
+
+class UserNoteCog(Cog, name="User notes"):
+    CONTRIBUTORS = [Contributor.Florian]
+
+    @commands.group()
+    async def user_note(self, ctx: Context):
+        """
+        mange notes for users
+        """
+
+        if ctx.invoked_subcommand is None:
+            raise UserInputError
+
+    @user_note.command(name="add")
+    @UserNotePermissions.write.check
+    async def add_user_note(self, ctx: Context, member: Member, *, message: str):
+        await UserNote.create(
+            member=member.id,
+            applicant=ctx.author.mention,
+            message=message,
+        )
+        await ctx.message.add_reaction(name_to_emoji["white_check_mark"])
+
+    @user_note.command(name="remove")
+    @UserNotePermissions.write.check
+    async def remove_user_note(self, ctx: Context, message_id: str):
+        user_notes = await db.get(UserNote, message_id=message_id)
+        if user_notes:
```

raise a CommandError instead of failing silently